pkg/istio-agent/xds_proxy.go

```
@@ -444,6 +449,18 @@ func (p *XdsProxy) handleUpstreamRequest(con *ProxyConnection) {
 				return
 			}
 
+			// override the first xds request node metadata labels
+			// TODO: if we want to regenerate all metadata, use `GetNodeMetaData`
+			if !initialRequestsSent.Load() {
+				if req.Node != nil {
+					meta, _ := model.ParseBootstrapNodeMetadata(req.Node.Metadata)
```

not sure we should ignore error here?